File: CppPractice/Project1/Project1/Leetcode118.cpp
#include "LeetcodeHeader.h"

class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> ans;
        vector<int> temp;
        if (numRows == 0)
        {
            return ans;
        }
        else if (numRows == 1)
        {
            temp = { 1 };
            ans.push_back(temp);
            return ans;
        }
        else if (numRows == 2)
        {
            temp = { 1 };
            ans.push_back(temp);
            temp = { 1,1 };
            ans.push_back(temp);
            return ans;

        }
        else
        {
            temp = { 1 };
            ans.push_back(temp);
            temp = { 1,1 };
            ans.push_back(temp);
            for (int i = 3; i <= numRows; i++)
            {
                temp = { };
                for (int j = 0; j < i; j++)
                {
                    if (j == 0 or j == i - 1)
                    {
                        temp.push_back(1);
                    }
                    else
                    {
                        temp.push_back(ans.back()[j - 1] + ans.back()[j]);
                    }
                }
                ans.push_back(temp);
            }

        }

        return ans;
    }
};

File: CSharpPractice/LeetcodePractice/Leetcode118.cs
public class Solution118 {
    public IList<IList<int>> Generate(int numRows) 
    {
        List<IList<int>> ans = new List<IList<int>>();
        List<int> temp = new List<int>{1};
        ans.Add(new List<int>(temp));
        if(numRows==1){return ans;}
        temp.Add(1);
        ans.Add(new List<int>(temp));
        if(numRows==2){return ans;}
        for(int i=0;i<numRows-2;i++)
        {
            List<int> newLine = new List<int>{1};
            for(int j=1;j<i+2;j++)
            {
                newLine.Add(temp[j-1]+temp[j]);

            }
            newLine.Add(1);
            temp = newLine;
            ans.Add(temp);
        }
        return ans;        
    }
}